adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -156,11 +168,12 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
   //   second.
   protected val optPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]]
 
-  assert(optPartitionMap.isEmpty ||
+  assert(optPartitionMap == null ||
```

Comment:
This happens due to some weird serialization behavior (IIRC related to the closure cleaner) that comes up in `pipe` since we use the `newRdd.getReferenceRegion` function inside of a `mapPartitionsWithIndex` call. I don't really have any great insight as to what we can do to fix it, but IIRC all the classes that extend `GenomicRDD` have constructors that are `private[rdd]`, so we should be able to just apply sanity in our code.